tests/job-spawner-and-cleaner.py

```diff
@@ -13,13 +13,18 @@
 
 if len(sys.argv) < 2:
     raise Exception("first command line argument must be redis host")
-db = redis.Redis(host=sys.argv[1].replace(":6379",""))
+
+host = sys.argv[1].split(":")
+
+if len(host) == 2:
+    db = redis.Redis(host=host[0], port=host[1])
+else:
+    db = redis.Redis(host=host[0])
 
 if len(db.keys("*")) > 0:
     raise Exception("redis database isn't clean")
 
-
-queue_list.append("shared_jobs")
+shared_jobs = WorkQueue(KeyPrefix("shared_jobs"))
 queue_list_names = []
 
 for index in range(len(queue_list)):
```

Comment:
Why copy `queue_list` to `queue_list_names` and overwrite `queue_list` when you could just make one new list and leave the names unmodified?